core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -491,6 +491,101 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
     }
   }
 
+  /**
+   * Set available resources on the driver/executor.
+   */
+  private[spark] def setResources(
+      resourceName: String,
+      resourceCount: Int,
+      resourceAddresses: Option[Seq[String]],
```

Comment:
this is not a general solution because executors might have different addresses.

Reply:
I agree, I'm not sure why we need this in SparkConf, if its just for testing, make it a testing util?